zummon/src/commonMain/kotlin/companysurvey/Validation.kt

```diff
@@ -202,6 +254,138 @@ class ElectricityValidator : Validator<Electricity> {
         }
     }
 
+    //annual feed-in should match total of quarter-hourly feed-in
+    fun validateAnnualFeedInMatchesQuarterHourlyFeedIn(electricity: Electricity): ValidationResult {
+        if (electricity.quarterHourlyFeedIn_kWh == null) {
+            return ValidationResult(Status.MISSING_DATA, translate("electricity.quarterHourlyFeedInNotProvided"))
+        }
+
+        return if (electricity.quarterHourlyFeedIn_kWh.hasFullYear() == true) {
+            val totalQuarterHourlyFeedIn = electricity.quarterHourlyFeedIn_kWh.values.sum()
+            if (electricity.annualElectricityFeedIn_kWh?.toFloat() != totalQuarterHourlyFeedIn) {
```

Comment:
It will never match exactly due to rounding and floating point artifacts. We should check that they're within like 1%.